Navigation Link: Add "Open in new tab" toggle to navigation block sidebar

Infinite-Null opened the pull request:
Fixes: #67246 

## What?
Adds the "Open in new tab" toggle option to the Navigation Link block's settings sidebar, making it accessible alongside other link settings.

## Why?
Currently, the "Open in new tab" setting is only available within the link modal of the context menu, while all other link options are presented in the block settings sidebar. This inconsistent placement has confused editors trying to locate this commonly used option.

## Screenshots or screencast

![After Patch (2)](https://github.com/user-attachments/assets/9d102fad-64e9-4026-9709-1dc9fe2c56c3)

Infinite-Null committed:
Navigation: Add "Open in new tab" toggle to navigation block sidebar

relic-se commented:
Yes, that looks great. I think you've provided a quick, simple solution for this problem. All clear with me!

Infinite-Null committed:
Navigation: Add "Open in New Tab" toggle under Link settings

Infinite-Null committed:
Merge branch 'trunk' into add/nav-links-target-option-to-sidebar

Infinite-Null committed:
Navigation Link: Move "Open in new tab" toggle into separate ToolsPanelItem

Infinite-Null committed:
Merge branch 'trunk' into add/nav-links-target-option-to-sidebar

Infinite-Null committed:
Merge branch 'trunk' into add/nav-links-target-option-to-sidebar

Infinite-Null committed:
Add 'Open in new tab' option to Navigation Link controls

Infinite-Null committed:
Reset unnecessary change

Infinite-Null committed:
Merge branch 'trunk' into add/nav-links-target-option-to-sidebar

t-hamano reviewed (commented):
Sorry for the late reply.

The code looks good, let's wait for design feedback.

![image](https://github.com/user-attachments/assets/aeb367a2-885c-4fe6-8dc6-41c3a71a9070)

Infinite-Null committed:
Merge branch 'trunk' into add/nav-links-target-option-to-sidebar

jasmussen commented:
Should it be a simple checkbox to match what's inside the link dialog?
![Screenshot 2025-06-30 at 10 44 05](https://github.com/user-attachments/assets/367d7de7-fefb-4070-90ef-4e51df4ae375)

It doesn't seem valuable to have two different designs for this. Thanks for the PR!

Infinite-Null committed:
Navigation: Replace ToggleControl with CheckboxControl for 'Open in new tab' option

Infinite-Null commented:
Hi @jasmussen,
Thanks for the feedback 🙌! I’ve updated the PR to use a simple checkbox to match what’s inside the link dialog, ensuring consistency in design. Please review it at your convenience.
### Screenshot:
![Untitled design](https://github.com/user-attachments/assets/518d4117-296a-46da-af3c-37dfad02a2e6)

jasmussen commented:
Nice quick action. I'd omit the help text, though, the checkbox itself seems plenty descriptive, and by removing the help text you further ensure it's the same design.

Infinite-Null committed:
Remove help text from CheckboxControl for 'Open in new tab' option

Infinite-Null commented:
Thanks for the feedback @jasmussen,
I have updated the PR and removed the help text:
![image](https://github.com/user-attachments/assets/137d0065-107a-4f60-a536-ace72c5038d6)

jasmussen commented:
Nice, works for me. I'll defer to others on the details. 👍 👍

getdave commented:
We should probably apply the same to the Submenu block

priethor reviewed (commented):
> We should probably apply the same to the Submenu block

We should check the interface of the Button block, as adding a link there is very different.
